### PR TITLE
fix(dashboard): Charts crashing when cross filter on adhoc column is applied

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
When there are charts using different datasets in a dashboard and a cross filter is applied on an adhoc column, the charts that use different dataset than the origin chart are crashing with a db error.
This PR fixes that by excluding filters based on an incompatible adhoc column from the query.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

image

After:

image

### TESTING INSTRUCTIONS
1. Create a chart that is grouped by an adhoc column (with sql expression) and add it to a dashboard
2. Add another chart to a dashboard, which should use a different dataset than the first chart
3. Apply a cross filter by clicking on a series related to the adhoc column in the first chart
4. Verify that the second chart doesn't crash and is not filtered by the cross filter

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/exceptions.py [hunk 1]
```diff
@@ -270,3 +270,7 @@ class SupersetCancelQueryException(SupersetException):
 
 class QueryNotFoundException(SupersetException):
     status = 404
+
+
+class ColumnNotFoundException(SupersetException):
```
Comment (eschutho) [inline superset/exceptions.py:275]: Should we make this a 404/not found status?

**Review by eschutho — CHANGES_REQUESTED** [review]
Can we write a test to cover this bug, too?

File: superset/exceptions.py (hunk 1 above)
Comment (kgabryje) [reply superset/exceptions.py:275]: Done

**Comment by kgabryje** [comment]
> Can we write a test to cover this bug, too?

Test added

File: tests/integration_tests/charts/data/api_tests.py [hunk 2]
```diff
@@ -73,6 +74,12 @@
     "when gender = 'girl' then 'female' else 'other' end",
 }
 
+INCOMPATIBLE_ADHOC_COLUMN_FIXTURE: AdhocColumn = {
+    "hasCustomLabel": True,
+    "label": "exciting_or_boring",
+    "sqlExpression": "case when genre = 'Action' then 'Exciting' " "else 'Boring' end",
```
Comment (villebro) [inline tests/integration_tests/charts/data/api_tests.py:80]: autoformatter nit:
```suggestion
    "sqlExpression": "case when genre = 'Action' then 'Exciting' else 'Boring' end",
```

**Review by villebro — APPROVED** [review]
Great work!

**Review by eschutho — APPROVED** [review]

**Comment by eschutho** [comment]
Thanks @kgabryje! Looks great

**Pull request merged**